crates/sui-graphql-rpc/src/extensions/timeout.rs

```diff
@@ -2,35 +2,69 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use async_graphql::{
-    extensions::{Extension, ExtensionContext, ExtensionFactory, NextRequest},
-    Response, ServerError,
+    extensions::{Extension, ExtensionContext, ExtensionFactory, NextExecute, NextParseQuery},
+    parser::types::ExecutableDocument,
+    Response, ServerError, ServerResult,
 };
-use std::sync::Arc;
+use async_graphql_value::Variables;
 use std::time::Duration;
-use tokio::time::timeout;
+use std::{net::SocketAddr, sync::Arc};
+use tokio::{sync::Mutex, time::timeout};
```

Comment:
I notice we're using tokio's `Mutex` here rather than the `std` one. Tokio has some [docs](https://doc.servo.org/tokio/sync/struct.Mutex.html#which-kind-of-mutex-should-you-use) on the pro's/con's.

Maybe this is something you've looked at but I wanted to flag because I think in this case the `std` version may be more appropriate?

Reply:
Thanks, I switched to std Mutex. I only used tokio as we're using in some other parts of the code...